src/main/java/com/google/step/datamanager/DealManagerDatastore.java

```diff
@@ -146,37 +149,54 @@ public Deal updateDeal(Deal deal, List<String> tagNames) {
     return transformEntityToDeal(dealEntity);
   }
 
-  /** Retrieves deals posted by restaurants or users */
-  private List<Deal> getDealsPublishedByRestaurantsOrUsers(
-      Set<Long> ids, String filterAttribute, int limit) {
-    List<Deal> dealResults = new ArrayList<>();
+  /** Retrieves deals with a filter applied */
+  private List<Long> getDealsWithFilter(
+      List<Object> ids, String filterAttribute, int limit, String sort) {
```

Comment:
what is the `sort` variable for? It seems like the function only check if sort is null

Reply:
its to sort based on any deal attribute, now its just sort based on the timestamp so i only checked if sort is null but if there is more to sort on then it can be if sort == "new" etc

Reply:
I don't think it's good practice to pass strings to functions to change the logic, because there's no meaning to the string, like whoever reading the code need to look inside the function and see what these "special strings" are, and is very messy. And passing in strings are also prone to typos.